pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -125,28 +125,52 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 		if err := r.addFinalizer(userAcc); err != nil {
 			return reconcile.Result{}, err
 		}
-
+	}
+	if !util.IsBeingDeleted(userAcc) && !userAcc.Spec.Disabled {
 		var createdOrUpdated bool
 		var user *userv1.User
 		if user, createdOrUpdated, err = r.ensureUser(reqLogger, userAcc); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-
 		if _, createdOrUpdated, err = r.ensureIdentity(reqLogger, userAcc, user); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-
 		if _, createdOrUpdated, err = r.ensureNSTemplateSet(reqLogger, userAcc); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-	} else if util.HasFinalizer(userAcc, userAccFinalizerName) {
+	}
+	if util.HasFinalizer(userAcc, userAccFinalizerName) || userAcc.Spec.Disabled {
```

Comment:
I think that the logic to handle the disabled case should go in a different `if` block, and that the status shouldn't be updated in the same reconcile loop as the identity and user deletions.  Once the identity and user resources are deleted, it will trigger another reconcile loop for the owning `UserAccount` resource, correct? If that's the case, then it should be this reconcile that updates the status.

Reply:
done here: e0c55d8218eff7e1c406842e0d44110caebaca49

Reply:
@sbryzak i have moved the the disabled part into another if block but i'm not sure it should stay this way. the reason we're waiting to set the status to disabled until the next reconcile is to ensure that they have both been deleted prior to setting the status. the spec is the desired state and the status is the actual state - hence we should wait until all conditions are met prior to setting the status.

Reply:
fixed!